src/hotspot/share/cds/heapShared.cpp

```diff
@@ -2096,6 +2125,18 @@ bool HeapShared::is_dumped_interned_string(oop o) {
   return _dumped_interned_strings->get(o) != nullptr;
 }
 
+// These tables should be used only within the CDS safepoint, so
+// delete them before we exit the safepoint. Otherwise the table will
+// contain bad oops after a GC.
```

Comment:
So is this just for good code hygiene? I don't see any changes that involve checking these table pointers for null, so presumably no code actually tries to access them outside the safepoint?

Reply:
Yes, this code is just for code hygiene for now. I expected it to be used when we integrate AOT code compilation.